### PR TITLE
feat: Add `log_spans` to client and REST API

### PR DESCRIPTION
**Pull request by anticorrelator**
- Adds `POST projects/{id}/spans` route and corresponding client method

Adds new `log_spans` and `log_spans_dataframe` client methods that enables creation of spans by submitting spans with the same structure returned by the `get_spans` method or `get_spans_dataframe` method. `log_spans` accepts a sequence of spans and will not queue spans if any spans are invalid for insertion (they might fail validation, or they are not unique)

Also adds `uniquify_spans` and `uniquify_spans_dataframe` helpers that regenerates the span- and trace- ids to ensure no conflicts on insertion.

```python
from phoenix.client import Client
from phoenix.client.helpers.spans import uniquify_spans

client = Client()

spans = [
    {
        "name": "llm_call",
        "context": {"trace_id": "trace_123", "span_id": "span_456"},
        "start_time": "2024-01-15T10:00:00Z",
        "end_time": "2024-01-15T10:00:05Z",
        "span_kind": "LLM"
    }
]

unique_spans = uniquify_spans(spans)
result = client.spans.log_spans(
    project_identifier="my-project",
    spans=unique_spans,
)
```

## Summary by Sourcery

Add end-to-end span creation support: implement a new create_spans client method and REST API route, provide ID regeneration helper, improve error handling, and add integration tests to cover span queuing scenarios.

New Features:
- Add create_spans method to sync and async clients for enqueuing spans via a new POST API
- Implement POST /projects/{project_identifier}/spans endpoint in the server API to accept and queue spans
- Introduce helper uniquify_spans to regenerate span and trace IDs while preserving relationships
- Introduce SpanCreationError exception in client to surface span queuing failures

Enhancements:
- Handle FastAPI 422 validation errors in create_spans, converting them to structured invalid_spans responses
- Log warnings and info summaries for invalid or duplicate spans when create_spans is called
- Adjust get_spans method to cast responses to the correct SpansResponseBody type

Documentation:
- Extend client docstrings with examples demonstrating create_spans usage, error handling, and regenerating IDs

Tests:
- Add comprehensive integration tests covering basic span creation, duplicate detection, invalid spans handling, batch creation, timeout and non-existent project errors
- Add tests for uniquify_spans utility to verify regenerated IDs and successful span insertion

### CODE REVIEW

File: packages/phoenix-client/src/phoenix/client/resources/spans/__init__.py [hunk 1]
```diff
@@ -808,6 +913,73 @@ async def get_spans(
 
         return all_spans[:limit]
 
+    async def log_spans(
+        self,
+        *,
+        project_identifier: str,
+        spans: Sequence[v1.Span],
```
Comment (mikeldking) [inline packages/phoenix-client/src/phoenix/client/resources/spans/__init__.py:920]: how easy is it to construct a v1.Span?

**Review by mikeldking — APPROVED** [review]
Looks good to me from a client method perspective. I like the dataframe and non dataframe roundtripping. It feels like a clean delineation is forming.

Handing off to @axiomofjoy or @RogerHYang for a quick look but I like the footprint.

File: src/phoenix/server/api/routers/v1/spans.py [hunk 2]
```diff
@@ -393,7 +412,9 @@ class SpanEvent(V1RoutesBaseModel):
 
 
 class Span(V1RoutesBaseModel):
-    id: str = Field(description="Span Global ID, distinct from the OpenTelemetry span ID")
+    id: str = Field(
+        default="", description="Span Global ID, distinct from the OpenTelemetry span ID"
+    )
     name: str = Field(description="Name of the span operation")
     context: SpanContext = Field(description="Span context containing trace_id and span_id")
     span_kind: str = Field(description="Type of work that the span encapsulates")
```
Comment (axiomofjoy) [inline src/phoenix/server/api/routers/v1/spans.py:420]: It looks like this is intended to allow re-use between create routes (which don't require an ID) and query routes (which have the ID). We have an existing convention to use `.*Data` to describe the type without the ID (e.g., `FreeformAnnotationConfig` vs. `FreeformAnnotationConfigData`)

File: packages/phoenix-client/src/phoenix/client/helpers/spans/__init__.py
Comment (axiomofjoy) [inline packages/phoenix-client/src/phoenix/client/helpers/spans/__init__.py:1]: Not sure I fully understand the usage of uniquify methods. How are the uniquified objects used downstream?

File: packages/phoenix-client/src/phoenix/client/exceptions/__init__.py
Comment (axiomofjoy) [inline packages/phoenix-client/src/phoenix/client/exceptions/__init__.py:1]: According to ChatGPT, it's possible to push exception types through the OpenAPI schema.

Untested snippet:

```python
my_custom_error_response = {
    418: {
        "model": MyCustomExceptionModel,
        "description": "My custom error",
    }
}

@app.get("/foo", responses=my_custom_error_response)
@app.get("/bar", responses=my_custom_error_response)
```

File: packages/phoenix-client/src/phoenix/client/resources/spans/__init__.py (hunk 1 above)
Comment (anticorrelator) [reply packages/phoenix-client/src/phoenix/client/resources/spans/__init__.py:920]: currently we get them from either `get_spans` or converting them from dataframes. Constructing them requires using our typedict for autocomplete, which can be a bit cumbersome. The most ergonomic path, currently is to have loaded a collection of v1.Spans from spans that were previously retrieved and saved as JSON

File: src/phoenix/server/api/routers/v1/spans.py (hunk 2 above)
Comment (anticorrelator) [reply src/phoenix/server/api/routers/v1/spans.py:420]: Discussed offline: we've unified the "creation payload" and the "return payload" for spans to improve user ergonomics. We want users to be able to the same object back and forth between the API
